docs/source/tutorials/06_using_your_own_models.rst

```diff
@@ -0,0 +1,578 @@
+castings based on the casting IDs.
+
+   .. figure:: /assets/tutorials/casting_predictions_viz.png
+      :width: 832
+      :alt: Casting prediction visualization
+
+      Casting Prediction Visualization
+
+This concludes the guided example on using your own custom models. 
+
+.. _tutorial_custom_object_detection_models:
+
+Custom Object Detection Models
+==============================
+
+The previous example was centered on the task of image classification. *Object detection* is
+another common task in Computer Vision. PeekingDuck offers several pre-trained
+:doc:`object detection </resources/01a_object_detection>` model nodes which can detect up to
+80 different types of objects, such as persons, cars, and dogs, just to name a few. For the
+complete list of detectable objects, refer to the 
+:ref:`Object Detection IDs <general-object-detection-ids>` page. Quite often, you may need to train 
+a custom object detection model on your own dataset, such as defects on a printed circuit board
+(PCB) as shown below. This section discusses some important considerations for the object detection
+task, supplementing the guided example above.
+
+   .. figure:: /assets/tutorials/PCB_defect.png
+      :width: 416
+      :alt: Object detection of defects on PCB
+
+      Object Detection of Defects on PCB (Source: `The Institution of Engineering and Technology 
+      <https://ietresearch.onlinelibrary.wiley.com/doi/10.1049/trit.2019.0019>`_)
+
+PeekingDuck's object detection model nodes conventionally receive the :term:`img` data type, and 
+produce the :term:`bboxes`, :term:`bbox_labels`, and :term:`bbox_scores` data types. An example of
+this can be seen in the API documentation for a node such as :mod:`model.efficientdet`. We 
+strongly recommend keeping to these data type conventions for your custom object detection node,
+ensuring that they adhere to the described format, e.g. :term:`img` is in BGR format, and
+:term:`bboxes` is a NumPy array of a certain shape.
+
+This allows you to leverage on PeekingDuck's ecosystem of existing nodes. For example, by ensuring
+that your custom model node receives :term:`img` in the correct format, you are able to use
+PeekingDuck's :mod:`input.visual` node, which can read from multiple visual sources such as a 
+folder of images or videos, an online cloud source, or a CCTV/webcam live feed. By ensuring that
+your custom model node produces :term:`bboxes` and :term:`bbox_labels` in the correct format, you
+are able to use PeekingDuck's :mod:`draw.bbox` node to draw bounding boxes and associated labels
+around the detected objects.
+
+By doing so, you would have saved a significant amount of development time, and can focus more on
+developing and finetuning your custom object detection model. This was just a simple example, and
+you can find out more about PeekingDuck's nodes from our :ref:`API Documentation <api_doc>`, and
+PeekingDuck's in-built data types from our :doc:`Glossary </glossary>`.
```

Comment:
We have been using "built-in" to describe these data types

Reply:
Fixed